Marlin/src/pins/lpc1769/pins_MKS_SGEN_L_V2.h

```diff
@@ -0,0 +1,365 @@
+#else
+  #define Z_MIN_PIN                        P1_25  // Z-
+  #define Z_MAX_PIN                        P1_24  // Z+
+#endif
+
+//
+// Z Probe (when not Z_MIN_PIN)
+//
+#ifndef Z_MIN_PROBE_PIN
+  #define Z_MIN_PROBE_PIN                  P1_24
+#endif
+
+//
+// Steppers
+//
+#define X_STEP_PIN                         P2_02
+#define X_DIR_PIN                          P2_03
+#define X_ENABLE_PIN                       P2_01
+#ifndef X_CS_PIN
+  #define X_CS_PIN                         P1_01
+#endif
+
+#define Y_STEP_PIN                         P0_19
+#define Y_DIR_PIN                          P0_20
+#define Y_ENABLE_PIN                       P2_08
+#ifndef Y_CS_PIN
+  #define Y_CS_PIN                         P1_08
+#endif
+
+#define Z_STEP_PIN                         P0_22
+#define Z_DIR_PIN                          P2_11
+#define Z_ENABLE_PIN                       P0_21
+#ifndef Z_CS_PIN
+  #define Z_CS_PIN                         P1_10
+#endif
+
+#define E0_STEP_PIN                        P2_13
+#define E0_DIR_PIN                         P0_11
+#define E0_ENABLE_PIN                      P2_12
+#ifndef E0_CS_PIN
+  #define E0_CS_PIN                        P1_15
+#endif
+
+#define E1_STEP_PIN                        P1_09
+#define E1_DIR_PIN                         P1_14
+#define E1_ENABLE_PIN                      P0_10
+#ifndef E1_CS_PIN
+  #define E1_CS_PIN                        P1_17
+#endif
+
+//
+// Software SPI pins for TMC2130 stepper drivers
+//
+#if ENABLED(TMC_USE_SW_SPI)
+  #ifndef TMC_SW_MOSI
+    #define TMC_SW_MOSI                    P1_16
+  #endif
+  #ifndef TMC_SW_MISO
+    #define TMC_SW_MISO                    P0_05
+  #endif
+  #ifndef TMC_SW_SCK
+    #define TMC_SW_SCK                     P0_04
+  #endif
+#endif
+
+#if HAS_TMC_UART
+  /**
+   * TMC2208/TMC2209 stepper drivers
+   *
+   * Hardware serial communication ports.
+   * If undefined software serial is used according to the pins below
+   */
+  //#define X_HARDWARE_SERIAL  Serial
+  //#define X2_HARDWARE_SERIAL Serial1
+  //#define Y_HARDWARE_SERIAL  Serial1
+  //#define Y2_HARDWARE_SERIAL Serial1
+  //#define Z_HARDWARE_SERIAL  Serial1
+  //#define Z2_HARDWARE_SERIAL Serial1
+  //#define E0_HARDWARE_SERIAL Serial1
+  //#define E1_HARDWARE_SERIAL Serial1
+  //#define E2_HARDWARE_SERIAL Serial1
+  //#define E3_HARDWARE_SERIAL Serial1
+  //#define E4_HARDWARE_SERIAL Serial1
+
+  //
+  // Software serial
+  //
+
+  #define X_SERIAL_TX_PIN                  P1_01
+  #define X_SERIAL_RX_PIN                  P1_01
+
+  #define Y_SERIAL_TX_PIN                  P1_08
+  #define Y_SERIAL_RX_PIN                  P1_08
+
+  #define Z_SERIAL_TX_PIN                  P1_10
+  #define Z_SERIAL_RX_PIN                  P1_10
+
+  #define E0_SERIAL_TX_PIN                 P1_15
+  #define E0_SERIAL_RX_PIN                 P1_15
+
+  #define E1_SERIAL_TX_PIN                 P1_17
+  #define E1_SERIAL_RX_PIN                 P1_17
+
+  #define Z2_SERIAL_TX_PIN                 P1_17
```

Comment:
Are the Z2 serial pins actually needed?

I assume that when E1 is used as a Z2, it will automatically use the E1 serial pins as well.